lib/SPIRV/SPIRVToOCL.h

```diff
@@ -185,4 +189,142 @@ class SPIRVToOCLLegacy : public ModulePass {
+public:
+  llvm::PreservedAnalyses run(llvm::Module &M,
+                              llvm::ModuleAnalysisManager &MAM) {
+    return runSPIRVToOCL(M) ? llvm::PreservedAnalyses::none()
+                            : llvm::PreservedAnalyses::all();
+  }
+};
+
+class SPIRVToOCL12Legacy : public SPIRVToOCL12Base, public SPIRVToOCLLegacy {
+public:
+  SPIRVToOCL12Legacy() : SPIRVToOCLLegacy(ID) {
+    initializeSPIRVToOCL12LegacyPass(*PassRegistry::getPassRegistry());
+  }
+  bool runOnModule(Module &M) override;
+
+  static char ID;
+};
+
+class SPIRVToOCL20Base : public SPIRVToOCLBase {
+public:
+  bool runSPIRVToOCL(Module &M) override;
+
+  /// Transform __spirv_MemoryBarrier to atomic_work_item_fence.
+  ///   __spirv_MemoryBarrier(scope, sema) =>
+  ///       atomic_work_item_fence(flag(sema), order(sema), map(scope))
+  void visitCallSPIRVMemoryBarrier(CallInst *CI) override;
+
+  /// Transform __spirv_ControlBarrier to work_group_barrier/sub_group_barrier.
+  /// If execution scope is ScopeWorkgroup:
+  ///    __spirv_ControlBarrier(execScope, memScope, sema) =>
+  ///         work_group_barrier(flag(sema), map(memScope))
+  /// Otherwise:
+  ///    __spirv_ControlBarrier(execScope, memScope, sema) =>
+  ///         sub_group_barrier(flag(sema), map(memScope))
+  void visitCallSPIRVControlBarrier(CallInst *CI) override;
+
+  /// Transform __spirv_Atomic* to atomic_*.
+  ///   __spirv_Atomic*(atomic_op, scope, sema, ops, ...) =>
+  ///      atomic_*(generic atomic_op, ops, ..., order(sema), map(scope))
+  Instruction *visitCallSPIRVAtomicBuiltin(CallInst *CI, Op OC) override;
+
+  /// Transform __spirv_OpAtomicIIncrement / OpAtomicIDecrement to
+  /// atomic_fetch_add_explicit / atomic_fetch_sub_explicit
+  Instruction *visitCallSPIRVAtomicIncDec(CallInst *CI, Op OC) override;
+
+  /// Conduct generic mutations for all atomic builtins
+  CallInst *mutateCommonAtomicArguments(CallInst *CI, Op OC) override;
+
+  /// Transform atomic builtin name into correct ocl-dependent name
+  Instruction *mutateAtomicName(CallInst *CI, Op OC) override;
+
+  /// Transform __spirv_OpAtomicCompareExchange/Weak into
+  /// compare_exchange_strong/weak_explicit
+  Instruction *visitCallSPIRVAtomicCmpExchg(CallInst *CI, Op OC) override;
+};
+
+class SPIRVToOCL20Pass : public llvm::PassInfoMixin<SPIRVToOCL20Pass>,
+                         public SPIRVToOCL20Base {
+public:
+  llvm::PreservedAnalyses run(llvm::Module &M,
+                              llvm::ModuleAnalysisManager &MAM) {
+    return runSPIRVToOCL(M) ? llvm::PreservedAnalyses::none()
+                            : llvm::PreservedAnalyses::all();
+  }
+};
+
+class SPIRVToOCL20Legacy : public SPIRVToOCLLegacy, public SPIRVToOCL20Base {
+public:
+  SPIRVToOCL20Legacy() : SPIRVToOCLLegacy(ID) {
+    initializeSPIRVToOCL20LegacyPass(*PassRegistry::getPassRegistry());
+  }
+  bool runOnModule(Module &M) override;
+  static char ID;
+};
+
 } // namespace SPIRV
+
+#endif // SPIRVTOOCL_H
```

Comment:
Please fix the missing EOL at the end of the file.

Reply:
Thanks, done!